### PR TITLE
Make sure the security user is from Contao

### PR DESCRIPTION
**Pull request by aschempp**
We must not handle the session for non-Contao users in Symfony.

### CODE REVIEW

**Comment by leofeyer** [comment]
Should be:

```php
if (!$this->hasUser() || !$this->isContaoMasterRequest($event)) {
    return;
}

$userObj = $this->getUserObject();

if (!$userObj instanceof User) {
    return;
}

$session = $this->getUserObject()->session;
…
```

**Comment by aschempp** [comment]
Would obviously be the same ;-)

**Comment by leofeyer** [comment]
No, my version only has one `getUserObject()` call, yours has two.

**Comment by aschempp** [comment]
I did read "shouldn't it be" instead of "should be" ^^
Changed it now.

**Comment by bytehead** [comment]
:+1:

**Comment by leofeyer** [comment]
You didn't remove it from line 64 though :)

File: src/EventListener/UserSessionListener.php [hunk 1]
```diff
@@ -84,6 +85,10 @@ public function onKernelResponse(FilterResponseEvent $event)
 
         $user = $this->getUserObject();
 
+        if (!$this->getUserObject() instanceof User) {
```
Comment (bytehead) [inline src/EventListener/UserSessionListener.php]: why not `if (!$user instanceof User) {` ?

Comment (leofeyer) [inline src/EventListener/UserSessionListener.php]: See https://github.com/contao/core-bundle/pull/361#issuecomment-148026917

**Comment by aschempp** [comment]
Sorry, my fault ^^. Fixed it

**Comment by bytehead** [comment]
It's still not what @leofeyer wants to have ^^

**Comment by leofeyer** [comment]
Merged in b196d1b0b353276207cc0a13ba9a873e0c2841ed.

**Comment by bytehead** [comment]
:+1:

**Pull request closed without merging**